geopandas/geodataframe.py

```diff
@@ -459,6 +459,17 @@ def __getitem__(self, key):
             result.__class__ = DataFrame
         return result
 
+
+    def explode(self):
+        """
+        Overriden from GeoPandasBase, returns rest of the dataframe joined to 
+        exploded geometry column.
+        """
+        exploded = self._get_geometry().explode().reset_index(level=1, drop=True)
```

Comment:
I think you can use `self.geometry` instead of `self._get_geometry()` ?

Reply:
Fixed.
I got a bit confused actually, how does `self.set_geometry` actually set `self.geometry`?

Reply:
you can check the `def set_geometry(..)` in this file, it basically replaces one column (the column with the geometry_column_name name)